zerver/views/streams.py

```diff
@@ -379,28 +384,42 @@ def update_stream_backend(
 
     for setting_name, permission_configuration in Stream.stream_permission_group_settings.items():
         request_settings_dict = locals()
```

Comment:
Should this line be outside the loop? Preexisting bug, and worth auditing to check if we make this mistake in other places, so can be a follow-up PR.